### PR TITLE
feat: aau i endorse user technical expertise - with vc

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
## Description
This PR is to apply useVerifiableCredential into AccountTEEndorsement
and handle the sign error message

### Related ticket

Fixes #

### Type of change

- [X] Chore
- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update

### CODE REVIEW

File: src/hooks/useVerifiableCredential.ts [hunk 1]
```diff
@@ -82,10 +82,12 @@ export function useVerifiableCredential() {
...
-          message: error.message,
-        });
+        if (error?.code !== 4001) {
+          setSignError({
+            type: VCSignErrorType.SignError,
+            message: error.message,
+          });
+        }
       }
       return null;
```
Comment (satyajeetkolhapure) [inline src/hooks/useVerifiableCredential.ts:92]: What happens when null is sent. e.g error code is 4001?
Will AccountTEEndorsement handle the null scenario?

Comment (stanleyyconsensys) [reply src/hooks/useVerifiableCredential.ts:92]: 4001 means user deined to sign in metamask

null case has handled in UI, will just keep stay in the modal, as deined to sign is not error or fail

File: src/pages/account/index.test.tsx [hunk 2]
```diff
@@ -13,6 +13,10 @@ jest.mock('../../features/account/components/AccountInfo', () => ({
   AccountInfo: () => <div />,
 }));
 
+jest.mock('../../features/account/AccountTEEndorsement', () => ({
```
Comment (satyajeetkolhapure) [inline src/pages/account/index.test.tsx:16]: minor : to test if mocked endorse component is present on index page.

File: src/hooks/useVerifiableCredential.ts (hunk 1 above)
Comment (satyajeetkolhapure) [reply src/hooks/useVerifiableCredential.ts:92]: In that case it's ok

File: src/pages/account/index.test.tsx (hunk 2 above)
Comment (stanleyyconsensys) [reply src/pages/account/index.test.tsx:16]: right, i just override the the whole endorsement components for easier test, but may be it is better to keep it for test if it present or not

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 3]
```diff
@@ -64,14 +109,14 @@ describe('AccountTEEndorsement', () => {
...
+    expect(queryByText('Software Security')).toBeInTheDocument();
+    expect(queryByText('Software Development')).toBeInTheDocument();
   });
 
-  it('assign ens name to `trustEntity` when `useEnsName` is return a name', async () => {
-    mockUseEnsName.mockImplementation(() => ({
-      data: 'mock.ens.name',
-      isLoading: false,
-    }));
+  it('assign trimed address to `trustEntity` when `useEnsName` is not ready or return null', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:117]:
```suggestion
  it('assigns trimed address to `trustEntity` when `useEnsName` is not ready or return null', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 4]
```diff
@@ -86,14 +131,12 @@ describe('AccountTEEndorsement', () => {
...
-    expect(queryByText('mock.ens.name')).toBeInTheDocument();
+    expect(queryByText(trimAddress(VALID_ACCOUNT_1))).toBeInTheDocument();
   });
 
-  it('dismiss the modal when sign button clicked', async () => {
-    mockUseEnsName.mockImplementation(() => ({
-      data: 'mock.ens.name',
-      isLoading: false,
-    }));
+  it('assign ens name to `trustEntity` when `useEnsName` is return a name', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:137]:
```suggestion
  it('assigns ens name to `trustEntity` when `useEnsName` returns a name', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx (hunk 3 above)
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:117]:
```suggestion
  it('assigns trimmed address to `trustEntity` when `useEnsName` is not ready or returns null', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 5]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
 
     expect(queryByText('mock.ens.name')).not.toBeInTheDocument();
   });
+
+  describe('when trigger onSign', () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:182]:
```suggestion
  describe('when onSign is triggered', () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 6]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+    it('calls `getSignedAssertion` when `signMessage` return signature', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:183]:
```suggestion
    it('calls `getSignedAssertion` when `signMessage` returns signature', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 7]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+          },
+          {
+            scope: TrustworthinessScope.SoftwareSecurity,
+            level: 0,
+          },
+        ],
+      );
+    });
+
+    it('does not calls `getSignedAssertion` when `signMessage` return null', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:214]:
```suggestion
    it('does not calls `getSignedAssertion` when `signMessage` returns null', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 8]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+      signMessageSpy.mockResolvedValue(null);
+
+      await runOnSign();
+
+      expect(signMessageSpy).toHaveBeenCalled();
+      expect(buildTechnicalExpertiseTrustSpy).toHaveBeenCalled();
+      expect(getSignedAssertionSpy).not.toHaveBeenCalled();
+    });
+
+    it('merge selected `ExpertiseTrust` and default `ExpertiseTrust`', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:232]:
```suggestion
    it('merges selected `ExpertiseTrust` and default `ExpertiseTrust`', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 9]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+            scope: TrustworthinessScope.SoftwareSecurity,
+            level: 0,
+          },
+        ],
+      );
+    });
+  });
+
+  describe('with signError', () => {
+    it('show `Failed to sign the message` toast message when signError is `SignError`', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:288]:
```suggestion
    it('shows `Failed to sign the message` toast message when signError is of type `SignError`', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 10]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+
+      expect(toastSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          title: 'Failed to sign the message',
+          description: 'sign error',
+        }),
+      );
+    });
+
+    it('show `Failed to verify signature` toast message when signError is `VerifyError`', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:313]:
```suggestion
    it('shows `Failed to verify signature` toast message when signError is of type `VerifyError`', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 11]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+
+      expect(toastSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          title: 'Failed to verify signature',
+          description: 'Your signature is invalid',
+        }),
+      );
+    });
+
+    it('show `Invalid Signature` toast message when signError is `SignError`', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:335]:
```suggestion
    it('shows `Invalid Signature` toast message when signError is of type `SignError`', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 12]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+
+      expect(toastSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          title: 'Invalid Signature',
+          description: 'invalid signature',
+        }),
+      );
+    });
+
+    it('show `Failed to sign the message` toast message when signError is unknown', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:360]:
```suggestion
    it('shows `Failed to sign the message` toast message when signError is unknown', async () => {
```

**Review by satyajeetkolhapure — APPROVED** [review]

File: src/features/account/AccountTEEndorsement.test.tsx (hunk 7 above)
Comment (ritave) [inline src/features/account/AccountTEEndorsement.test.tsx:214]:
```suggestion
    it('does not call `getSignedAssertion` when `signMessage` returns null', async () => {
```

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 13]
```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
...
+      signMessageSpy.mockResolvedValue('0xSignature');
+
+      const { getByText } = render(
+        <AccountTEEndorsement
+          address={VALID_ACCOUNT_1}
+          connectedAddress={VALID_ACCOUNT_2}
+        />,
+      );
+
+      await act(async () =>
```
Comment (ritave) [inline src/features/account/AccountTEEndorsement.test.tsx:250]: Why there's a need for double nested `act`s?

**Review by ritave — APPROVED** [review]

File: src/features/account/AccountTEEndorsement.test.tsx (hunk 13 above)
Comment (stanleyyconsensys) [reply src/features/account/AccountTEEndorsement.test.tsx:250]: it is to wait for the state update

Comment (stanleyyconsensys) [reply src/features/account/AccountTEEndorsement.test.tsx:250]: i find that the double act has apply everywhere, let me create another fix PR to fix them all in once

**Review by dawnseeker8 — APPROVED** [review]

**Pull request merged**